Extensions/BBText/JsExtension.js

```diff
@@ -0,0 +1,601 @@
+          .addExtraInfo('left')
+          .addExtraInfo('center')
+          .addExtraInfo('right')
+          .setLabel(_('Base alignment'))
+      );
+
+      objectProperties.set(
+        'fontFamily',
+        new gd.PropertyDescriptor(objectContent.fontFamily)
+          .setType('string')
+          .setLabel(_('Base font family'))
+      );
+
+      objectProperties.set(
+        'wordWrap',
+        new gd.PropertyDescriptor(objectContent.wordWrap ? 'true' : 'false')
+          .setType('boolean')
+          .setLabel(_('Word wrapping'))
+      );
+
+      objectProperties.set(
+        'visible',
+        new gd.PropertyDescriptor(objectContent.visible ? 'true' : 'false')
+          .setType('boolean')
+          .setLabel(_('Visible on start'))
+      );
+
+      return objectProperties;
+    };
+    objectBBText.setRawJSONContent(
+      JSON.stringify({
+        text:
+          '[b]bold[/b] [i]italic[/i] [size=15]smaller[/size] [font=times]times[/font] font\n[spacing=12]spaced out[/spacing]\n[outline=yellow]outlined[/outline] [shadow=red]DropShadow[/shadow] ',
+        opacity: 255,
+        fontSize: '20',
+        visible: true,
+        color: '#000000',
+        fontFamily: 'Arial',
+        align: 'left',
+        wordWrap: true,
+      })
+    );
+
+    objectBBText.updateInitialInstanceProperty = function(
+      objectContent,
+      instance,
+      propertyName,
+      newValue,
+      project,
+      layout
+    ) {
+      return false;
+    };
+    objectBBText.getInitialInstanceProperties = function(
+      content,
+      instance,
+      project,
+      layout
+    ) {
+      var instanceProperties = new gd.MapStringPropertyDescriptor();
+      return instanceProperties;
+    };
+
+    const object = extension
+      .addObject(
+        'BBText',
+        _('BBText'),
+        _(
+          'Displays a rich text label using BBCode markup (allowing to set parts of the text as bold, italic, use different colors and shadows).'
+        ),
+        'JsPlatform/Extensions/bbcode32.png',
+        objectBBText
+      )
+      .setIncludeFile('Extensions/BBText/bbtextruntimeobject.js')
+      .addIncludeFile('Extensions/BBText/bbtextruntimeobject-pixi-renderer.js')
+      .addIncludeFile('Extensions/BBText/dist/pixi-multistyle-text.umd.js');
+
+    // Utility function to add both a setter and a getter to a property from a list. Useful for setting multiple generic properties
+    const addSettersAndGettersToObjectHelper = (
+      gdObject,
+      properties,
+      objectName
+    ) => {
+      properties.forEach(property => {
+        property.type = property.type === 'boolean' ? 'yesorno' : property.type;
+
+        // Add all the generic GETTERS
+        if (property.type === 'number') {
+          gdObject
+            .addExpression(
+              `Get${property.functionName}`,
+              _(`Get ${property.description}`),
+              _(`Get ${property.description}`),
+              _(''),
```

Comment:
You don't put a sentence? Or don't know how to generate it?

Reply:
do you mean lines 198? I was hoping to get some suggestions on the wording on that. Less text in my mind leads to less reading which is a good thing :) 
if its line 199, I left it empty, because I didn't like how gdevelop pushes all custom added categories of the extension to the bottom of the list, so what is the extension's primary functionality is added in a way that requires the user scrolls every time to get to it. So I left them outside of a category to get them listed on the top

Reply:
Oops yes I was confused and thought I commented on the action/conditions :) Empty category is fine.